modules/pam_faildelay/pam_faildelay.c

```diff
@@ -94,14 +94,14 @@ int pam_sm_authenticate(pam_handle_t *pamh, int flags UNUSED,
 			int argc, const char **argv)
 {
     int i, debug_flag = 0;
-    long delay = -1;
+    long long delay = -1;
 
     /* step through arguments */
     for (i = 0; i < argc; i++) {
 	const char *val = pam_str_skip_prefix (argv[i], "delay=");
 	if (val != NULL) {
 	  delay = parse_delay (val);
-	  if (delay < 0 || (unsigned long) delay > UINT_MAX)
+	  if (delay < 0 || (unsigned long long) delay > UINT_MAX)
```

Comment:
The `delay < 0` condition is now redundant, but I suggest keeping it here anyway and let the compiler optimize this.

Reply:
Ok, I'll keep it. But thanks for pointing out!